datameta/linting.py

```diff
@@ -103,3 +106,84 @@ def lint_pending_msets(request, user, mset_ids = None):
     # report for the failed records
     good_sets = [ mdataset for mdataset in mdatasets if not linting_report[mdataset.id] ]
     return good_sets, file_pairs, linting_report
+
+
+def validate_metadataset_fields(
+    request:Request,
+    fields:dict, 
+    return_err_message:bool=False
+):
+    """Validate single metadataset in isolation"""
+    errors = [] # list of error
+                # empty means success
+
+    # get all metadatum fields:
+    db = request.dbsession
+    mdats_query = db.query(MetaDatum).order_by(MetaDatum.order).all()
+    mdats = { mdat.name: mdat for mdat in mdats_query }
+    
+    for name, value in fields.items():
+        # check if fields appeards in the metadatum list
+        if not name in mdats.keys():
+            errors.append({
+                "message": (
+                    f"field with name \"name\""
+                    "is not allowed"
+                ),
+                "field": name
+            })
+            continue
```

Comment:
Why don't we collect all extraneous fields here?

Reply:
What do you mean by that?

If there is an error with one field, the error will be captured and the loop moves on to the next field/value.